chargeio/services/Merchant/MerchantService.cs

```diff
@@ -1,16 +1,16 @@
-﻿using ChargeIo.Infrastructure;
-using ChargeIo.Models;
+﻿using ChargeIO.Infrastructure;
+using ChargeIO.Models;
 
-namespace ChargeIo.Services.Merchant
+namespace ChargeIO.Services.Merchant
 {
 	public class MerchantService
 	{
-        private string SecretKey { get; set; }
+        private string SecretKey { get; }
         
-        public MerchantService(string secretKey = null)
-		{
-            SecretKey = secretKey ?? Configuration.SecretKey;
-		}
+        public MerchantService(string secretKey = "")
+        {
+            SecretKey = secretKey.Length > 0 ? secretKey : Configuration.SecretKey;

```

Comment:
Ideally there would be some equivalent to the Java "@NonNull" annotation that would prevent a developer from passing null as an argument (because seriously, what good purpose does doing that even serve for anyone?) but in lieu of that I wrapped the logic up in a null check in e45261315420ce6d33d766be90627bdca31eca89.